python/lsst/ap/association/diaPipe.py

```diff
@@ -383,8 +383,9 @@ def run(self,
                                        inplace=True)
 
         # Append new DiaObjects and DiaSources to their previous history.
-        diaObjects = loaderResult.diaObjects.append(
-            createResults.newDiaObjects.set_index("diaObjectId", drop=False),
+        diaObjects = pd.concat(
```

Comment:
I'm very worried about using concat in our code like this: it can change the schema out from under us. Should we be using [convert_dtypes](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.convert_dtypes.html) here (and anywhere we use concat)? Could we move this away from pandas entirely and use `numpy.concatenate` or `numpy.vstack`?

If we had a way to get `pd.concat` to `raise` instead of silently casting to float, that might be more reasonable, but I don't think there is.

Reply:
I don't know anything about `convert_dtypes`, I just followed the instructions for how to phase out `append`. As for using pandas in general, see [my comment](https://github.com/lsst/ap_association/pull/160#issuecomment-1290868299) on #160.